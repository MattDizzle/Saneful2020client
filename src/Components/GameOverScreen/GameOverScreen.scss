.game-over-screen{
  display: flex;
  flex-direction: column;
  width: 100vw;
  max-width: 520px;
  height: 105vw;
  max-height: 540px;
  justify-content: center;
  align-items: center;
  z-index: 12;
  position: absolute;
  top: 0px; 
  left: 0px;
  background-color: rgb(0, 0, 0);
  color: rgb(255, 255, 255);
  padding: 8px;
}

.game-over-message{
  width: 80%;
}

.game-over-message p {
  line-height: 30px;
}