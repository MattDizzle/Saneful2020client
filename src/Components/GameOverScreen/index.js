import React from 'react';

import './GameOverScreen.scss';

const DialogBox = (props) => {

  const renderMessage = () => {

    let deathMessage = '';

    if (props.reason === 'sanity')
      deathMessage = 'you became lost in your own insanity';
    if (props.reason === 'health')
      deathMessage = 'your body degraded and you faded away';

    let timeLeft = props.currentTime;

    const numDays = Math.floor(timeLeft / 1440);
    timeLeft -= (1440 * numDays);
    const numHours = Math.floor(timeLeft / 60);
    timeLeft -= (60 * numHours);
    const numMins = timeLeft;

    return (
      <div className='game-over-message'>
        {/* <p>You lasted {(numDays===1)?`${numDays} day `:' days '}{(numHours===1)?`${numHours} hour`:' hours '} and {(numMins===1)?`${numMins} minute `:' minutes '} before you {deathMessage}...</p> */}
        <p>{`You lasted ${numDays} days ${numHours} hours and ${numMins} minutes before you ${deathMessage}...`}</p>
      </div>
    );

  };

  return (
    <div className='game-over-screen'>
      {renderMessage()}
    </div>
  );
};

export default DialogBox;