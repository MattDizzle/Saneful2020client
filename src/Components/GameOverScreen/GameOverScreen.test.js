import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import GameOverScreen from './index';


describe(`GameOverScreen Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <GameOverScreen />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <GameOverScreen />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})