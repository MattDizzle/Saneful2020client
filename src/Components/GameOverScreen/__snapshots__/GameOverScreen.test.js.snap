// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverScreen Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="game-over-screen"
>
  <div
    className="game-over-message"
  >
    <p>
      You lasted NaN days NaN hours and NaN minutes before you ...
    </p>
  </div>
</div>
`;
